src/DynamoCoreWpf/ViewModels/Menu/PreferencesViewModel.cs

```diff
@@ -433,14 +433,21 @@ public string SelectedPythonEngine
         {
             get
             {
-                return preferenceSettings.DefaultPythonEngine;
+                return selectedPythonEngine;
             }
             set
             {
-                if (value != preferenceSettings.DefaultPythonEngine)
+                if (value != selectedPythonEngine)
                 {
                     selectedPythonEngine = value;
-                    preferenceSettings.DefaultPythonEngine = value;
+                    if(value != Res.DefaultPythonEngineNone)
+                    {
+                        preferenceSettings.DefaultPythonEngine = value;
+                    }
+                    else{
+                        preferenceSettings.DefaultPythonEngine = string.Empty;
```

Comment:
Does this affect deserialization? Could you take some testing？

Reply:
Oh I see below you are actually checking the value so I assume you have done that :)